src/qibo/transpiler/unroller.py

```diff
@@ -85,7 +85,7 @@ def __call__(self, circuit: Circuit):
         Returns:
             (:class:`qibo.models.circuit.Circuit`): equivalent circuit with native gates.
         """
-        translated_circuit = circuit.__class__(circuit.nqubits)
+        translated_circuit = Circuit(**circuit.init_kwargs)
```

Comment:
this might be a problem if in the future more `kwargs` are added, no?

Reply:
I think this is better this way as the translated circuit should keep all the properties of the original circuit like density_matrix, qubit_names ecc...
I all the other steps of the transpiler i create a new circuit passing the init_kwargs